public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js

```diff
@@ -1,29 +1,65 @@
 /**
  * WordPress dependencies
  */
+import { __ } from '@wordpress/i18n';
 import { Disabled } from '@wordpress/components';
+import { decodeEntities } from '@wordpress/html-entities';
 
 /**
  * Internal dependencies
  */
-import FavoriteButton from '../favorite-button';
-import CopyPatternButton from '../copy-pattern-button';
 import Canvas from './canvas';
+import CopyPatternButton from '../copy-pattern-button';
+import FavoriteButton from '../favorite-button';
+import IconHeartFilled from '../icons/heart-filled';
 
-function PatternThumbnail( { pattern } ) {
+function getStatusLabel( pattern ) {
+	switch ( pattern.status ) {
+		case 'pending':
+			return __( 'Pending', 'wporg-patterns' );
+		case 'draft':
+			return __( 'Draft', 'wporg-patterns' );
+		case 'declined':
+			return __( 'Declined', 'wporg-patterns' );
+	}
+	return '';
+}
+
+function PatternThumbnail( { pattern, showAvatar } ) {
+	const statusLabel = getStatusLabel( pattern );
 	return (
 		<div className="pattern-grid__pattern">
-			<a href={ pattern.link } rel="bookmark">
-				<span className="screen-reader-text">{ pattern.title.rendered }</span>
-				<Disabled>
-					<Canvas className="pattern-grid__preview" html={ pattern.content.rendered } />
-				</Disabled>
-			</a>
-			<div className="pattern-grid__actions">
-				<h2 className="pattern-grid__title">{ pattern.title.rendered }</h2>
-				<FavoriteButton showLabel={ false } patternId={ pattern.id } />
-				<CopyPatternButton isSmall={ true } content={ pattern.pattern_content } />
+			<div className="pattern-grid__pattern-frame">
+				<a href={ pattern.link } rel="bookmark">
+					<span className="screen-reader-text">{ decodeEntities( pattern.title.rendered ) }</span>
+					<Disabled>
+						<Canvas className="pattern-grid__preview" html={ pattern.content.rendered } />
+					</Disabled>
+				</a>
+				{ statusLabel ? (
+					<div className={ `pattern-grid__status is-${ pattern.status }` }>
+						<span>{ statusLabel }</span>
+					</div>
+				) : null }
+				<div className="pattern-grid__actions">
+					<FavoriteButton showLabel={ false } patternId={ pattern.id } />
+					<CopyPatternButton isSmall={ true } content={ pattern.pattern_content } />
+				</div>
 			</div>
+
+			<h2 className="pattern-grid__title">{ decodeEntities( pattern.title.rendered ) }</h2>
+			<p className="pattern-grid__meta">
+				{ showAvatar && pattern.author_avatar ? (
```

Comment:
Should there be a default case for missing authors... if that case is possible??